execution/gethexec/espresso_finality_node.go

```diff
@@ -0,0 +1,158 @@
+package gethexec
+
+import (
+	"context"
+	"fmt"
+	espressoClient "github.com/EspressoSystems/espresso-sequencer-go/client"
+	espressoTypes "github.com/EspressoSystems/espresso-sequencer-go/types"
+	"github.com/ethereum/go-ethereum/arbitrum_types"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/offchainlabs/nitro/arbos"
+	"time"
+
+	"github.com/offchainlabs/nitro/arbos/arbostypes"
+	"github.com/offchainlabs/nitro/arbos/l1pricing"
+	"github.com/offchainlabs/nitro/util/stopwaiter"
+)
+
+var (
+	retryTime = time.Second * 1
+)
+
+type HotsShotState struct {
+	client          espressoClient.Client
+	nextSeqBlockBum uint64
```

Comment:
I think we should also persist this number in case the node lose this after a reboot

Reply:
I was thinking about it - if this finality node crashes someone can  set the `nextSeqBlockNum` to the number at which it crashed. 

On the other hand, if we store it in the db. If someone wants to start listening from a block later or before the `block number` at which the crash happened, they wont be able to do the same. Let me know what your thoughts are on the same!

Reply:
Thanks. Make sense. We can persist this number but it is not really necessary. Let's achieve the escape hatch in a simpler way first